### PR TITLE
Overridden Equals on wrapping Fake causes call to be forwarded to wrapped object

### PR DESCRIPTION
**Pull request by blairconrad**

### CODE REVIEW

**Comment by blairconrad** [comment]
Found while investigating performance improvements around resolving `Object` methods. `InterceptedFakeObjectCallExtensions` didn't look right.

File: tests/FakeItEasy.Specs/WrappingFakeSpecs.cs [hunk 1]
```diff
@@ -208,6 +208,44 @@ public static void FakeEqualsWrappedObjectWithValueSemantics(Bar realObject, IBa
...
+                .x(() => realObject = new ClassThatOverrideEqualsForNoReason());
+
+            "And a faked Object type that wraps the real object"
+                .x(() => wrapper = A.Fake<object>(o => o.Wrapping(realObject)));
+
+            "When Equals is called on the fake with itself as the argument"
+                .x(() => equals = wrapper.Equals(wrapper));
+
+            "Then it should return true"
+                .x(() => equals.Should().BeTrue());
```
Comment (blairconrad) [inline tests/FakeItEasy.Specs/WrappingFakeSpecs.cs:246]: This passed before the change to `InterceptedFakeObjectCallExtensions.cs`.

File: tests/FakeItEasy.Specs/WrappingFakeSpecs.cs [hunk 2]
```diff
@@ -208,6 +208,44 @@ public static void FakeEqualsWrappedObjectWithValueSemantics(Bar realObject, IBa
...
+                .x(() => realObject = new ClassThatOverrideEqualsForNoReason());
+
+            "And a faked class of the same type that wraps the real object"
+                .x(() => wrapper = A.Fake<ClassThatOverrideEqualsForNoReason>(o => o.Wrapping(realObject)));
+
+            "When Equals is called on the fake with itself as the argument"
+                .x(() => equals = wrapper.Equals(wrapper));
+
+            "Then it should return true"
+                .x(() => equals.Should().BeTrue());
```
Comment (blairconrad) [inline tests/FakeItEasy.Specs/WrappingFakeSpecs.cs:227]: This failed before the change to InterceptedFakeObjectCallExtensions.cs.

**Review by thomaslevesque — APPROVED** [review]
Thanks @blairconrad, looks fine to me!

**Pull request merged**